datalad/distribution/install.py

```diff
@@ -310,90 +162,144 @@ def __call__(
-        if path and isinstance(path, list):
-            if len(path) > 1:
-                if source is not None:
-                    raise ValueError("source argument not valid when "
-                                     "installing multiple datasets.")
-                else:
-                    for p in path:
-                        try:
-                            result = Install.__call__(
-                                path=p,
-                                source=None,
-                                dataset=dataset,
-                                get_data=get_data,
+        path = assure_list(path)
+
+        if not source and not path:
+            raise InsufficientArgumentsError(
+                "Please provide at least a source or a path")
+
+
+        ## Common kwargs to pass to underlying git/install calls.
+        #  They might need adjustments (e.g. for recursion_limit, but
+        #  otherwise would be applicable throughout
+        #
+        # There should have been more of common options!
+        # since underneath get could do similar installs, but now they
+        # have duplicated implementations which differ (e.g. get does not
+        # annex init installed annexes)
+        common_kwargs = dict(
+            get_data=get_data,
+            recursive=recursive,
+            recursion_limit=recursion_limit,
+            git_opts=git_opts,
+            annex_opts=annex_opts,
+            reckless=reckless,
+            jobs=jobs,
+        )
+
+        # switch into scenario without --source:
+        if source is None:
+            # we need to collect URLs and paths
+            to_install = []
+            to_get = []
+            for urlpath in path:
+                ri = RI(urlpath)
+                (to_get if isinstance(ri, PathRI) else to_install).append(urlpath)
+
+            installed_items = []
+            common_kwargs['dataset'] = dataset
+
+            # first install, and then get
+            for s in to_install:
+                lgr.debug("Install passes into install source=%s", s)
+                result = Install.__call__(
+                                source=s,
                                 description=description,
-                                recursive=recursive,
-                                recursion_limit=recursion_limit,
-                                save=save,
                                 if_dirty=if_dirty,
-                                git_opts=git_opts,
+                                save=save,
                                 git_clone_opts=git_clone_opts,
-                                annex_opts=annex_opts,
-                                annex_init_opts=annex_init_opts
+                                annex_init_opts=annex_init_opts,
+                                **common_kwargs
                             )
-
-                            installed_items += assure_list(result)
-                        except Exception:
```

Comment:
thanks! got it!  fixed now, although became again more involved ;): de62261717e9a64f90d79c6b9406cb65e43e271e